go-controller/hybrid-overlay/pkg/controller/master_test.go

```diff
@@ -93,7 +93,7 @@ var _ = Describe("Hybrid SDN Master Operations", func() {
 			stopChan := make(chan struct{})
 			f, err := factory.NewWatchFactory(fakeClient, stopChan)
 			Expect(err).NotTo(HaveOccurred())
-			defer close(stopChan)
+			defer f.Shutdown(stopChan)
```

Comment:
This is weirdly mixing semantics; if you have to call a special shutdown method then there's no reason to have a passed-in `stopChan`. Just have `NewWatchFactory` create the channel itself and store it in the factory, and have `f.Shutdown` take no args.

Reply:
done